cmd/create/ocmrole/cmd.go

```diff
@@ -326,52 +356,95 @@ func buildCommands(prefix string, roleName string, rolePath string, permissionsB
 	}
 	createRole := builder.Build()
 
-	createPolicy := awscb.NewIAMCommandBuilder().
-		SetCommand(awscb.CreatePolicy).
-		AddParam(awscb.PolicyName, policyName).
-		AddParam(awscb.PolicyDocument, fmt.Sprintf("file://sts_%s_permission_policy.json", aws.OCMRolePolicyFile)).
-		AddTags(iamTags).
-		AddParam(awscb.Path, rolePath).
-		Build()
+	var createPolicy string
+	if !managedPolicies {
+		createPolicy = awscb.NewIAMCommandBuilder().
+			SetCommand(awscb.CreatePolicy).
+			AddParam(awscb.PolicyName, policyName).
+			AddParam(awscb.PolicyDocument, fmt.Sprintf("file://sts_%s_permission_policy.json", aws.OCMRolePolicyFile)).
+			AddTags(iamTags).
+			AddParam(awscb.Path, rolePath).
+			Build()
+	}
 
+	var policyARN string
+	var err error
+	if managedPolicies {
+		policyARN, err = aws.GetManagedPolicyARN(policies, "sts_ocm_permission_policy")
+		if err != nil {
+			return "", err
+		}
+	} else {
+		policyARN = aws.GetPolicyARN(accountID, roleName, rolePath)
+	}
 	attachRolePolicy := awscb.NewIAMCommandBuilder().
 		SetCommand(awscb.AttachRolePolicy).
 		AddParam(awscb.RoleName, roleName).
-		AddParam(awscb.PolicyArn, aws.GetPolicyARN(accountID, roleName, rolePath)).
+		AddParam(awscb.PolicyArn, policyARN).
 		Build()
 
-	commands = append(commands, createRole, createPolicy, attachRolePolicy)
+	if managedPolicies {
+		commands = append(commands, createRole, attachRolePolicy)
+	} else {
+		commands = append(commands, createRole, createPolicy, attachRolePolicy)
+	}
 	if isAdmin {
 		policyName := aws.GetAdminPolicyName(roleName)
 
-		createAdminPolicy := awscb.NewIAMCommandBuilder().
-			SetCommand(awscb.CreatePolicy).
-			AddParam(awscb.PolicyName, policyName).
-			AddParam(awscb.PolicyDocument, fmt.Sprintf("file://sts_%s_permission_policy.json", aws.OCMAdminRolePolicyFile)).
-			AddTags(adminTags).
-			AddParam(awscb.Path, rolePath).
-			Build()
+		var createAdminPolicy string
+		if !managedPolicies {
+			createAdminPolicy = awscb.NewIAMCommandBuilder().
+				SetCommand(awscb.CreatePolicy).
+				AddParam(awscb.PolicyName, policyName).
+				AddParam(awscb.PolicyDocument, fmt.Sprintf("file://sts_%s_permission_policy.json", aws.OCMAdminRolePolicyFile)).
+				AddTags(adminTags).
+				AddParam(awscb.Path, rolePath).
+				Build()
+		}
 
+		if managedPolicies {
+			policyARN, err = aws.GetManagedPolicyARN(policies,
+				fmt.Sprintf("sts_%s_permission_policy", aws.OCMRolePolicyFile))
```

Comment:
Is there an OCMAdmin Policy for managed policies?

Reply:
Good catch!
Fixed, thanks.